envconfig.go

```diff
@@ -184,7 +184,12 @@ func gatherInfo(prefix string, spec interface{}, env map[string]string, isInside
 				}
 			}
 
-			f.Set(reflect.MakeSlice(f.Type(), l, l))
+			if l != 0 {
+				f.Set(reflect.MakeSlice(f.Type(), l, l))
+			} else {
+				l = f.Len()
+			}
```

Comment:
I think that we should extend the existing slice if `l > f.Len()`, wdyt? We could append some empty structs to it and process all env vars.